### DISCUSSION TITLE
Passing on a pipe via Process, doesn't execute the command, either it can be executed perfectly via command line on shell

### DISCUSSION THREAD

**Discussion by lmogthb** [symfony/symfony#46292] [Q&A] [answered]
Hello guys, I have experienced this error when I want to execute a command that has a pipe on it

This is the command that is perfectly executed on shell:

`grep "^ignoreip = " /etc/fail2ban/jail.d/defaults-debian.conf | cut -d "=" -f2`

And as Process Documentation says that we have to execute it passing on all the commands via array, I setup it like this

`$whiteList = new Process(['grep', '^ignoreip = ', '/etc/fail2ban/jail.d/defaults-debian.conf', '|', 'cut', '-d', '=', '-f2']);`

Have tried too many combinations but still can't get it to work, is there any way to get it working?

With shell_exec and exec, it works perfectly but can't get it to work with Process that is a better method on security

Thanks to all in advance

**Comment by stof** [comment, marked as answer]
`grep "^ignoreip = " /etc/fail2ban/jail.d/defaults-debian.conf | cut -d "=" -f2` is not 1 command. It is one command with its output piped into a second command, because of using a shell operator (the `|`).
This cannot be executed without a shell wrapper for the execution. So you will need to use `Process::fromSchellCommandLine` to execute that.

**Reply by lmogthb** [reply]
Oh thanks, it worked like a charm
